### PR TITLE
return real value as FC phase

### LINKED ISSUE #2773: CmdDef always returns positive FC phase.
### Information

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?
When we `.get` u1, u2 or u3 from `CmdDef`, all FrameChange phases become positive value.

### Steps to reproduce the problem
```
cmd_def = pulse.CmdDef.from_defaults(defaults.cmd_def, defaults.pulse_library)
cmd_def.get('u1', 0, P0=-pi)
```
this returns a schedule consisting of FrameChange of +pi (because P0 is negated in cmd_def).

```
cmd_def = pulse.CmdDef.from_defaults(defaults.cmd_def, defaults.pulse_library)
cmd_def.get('u1', 0, P0=pi)
```
this returns a schedule consisting of FrameChange of +pi, but this should be -pi.

### PR DESCRIPTION
**Pull request by nkanazawa1989**
### Summary
Fix #2773

### CODE REVIEW

File: qiskit/qobj/converters/pulse_instruction.py [hunk 1]
```diff
@@ -345,8 +347,10 @@ def convert_frame_change(self, instruction):
             phase_expr = parse_string_expr(phase, partial_binding=False)
 
             def gen_fc_sched(*args, **kwargs):
-                phase = abs(phase_expr(*args, **kwargs))
-                return commands.FrameChange(phase)(channel) << t0
+                _phase = phase_expr(*args, **kwargs)
```
Comment (taalexander) [inline qiskit/qobj/converters/pulse_instruction.py]: Does this often return a complex value? If it does not we should remove the check and just let this error be caught at a later time by allow Python duck typing to work.

Comment (nkanazawa1989) [reply qiskit/qobj/converters/pulse_instruction.py]: I don't think so. Usually parser returns a real value when imaginary part is zero. It returns complex value only when backends intentionally assign complex value as a phase, but this is invalid configuration.

Comment (nkanazawa1989) [reply qiskit/qobj/converters/pulse_instruction.py]: We can do this in marshmallow as well.

Comment (taalexander) [reply qiskit/qobj/converters/pulse_instruction.py]: So what is the path forward do you think?

Comment (nkanazawa1989) [reply qiskit/qobj/converters/pulse_instruction.py]: I realized real value is guaranteed by marshmallow.
https://github.com/Qiskit/qiskit-terra/blob/27b35555f3342ec6bbdedacf7cc1e2daf58961fe/qiskit/qobj/models/pulse.py#L56

If a user try to bundle complex value with `cmd_def.get`, marshmallow will raise an error during qobj generation. So I can simply remove type checking here.

Comment (taalexander) [reply qiskit/qobj/converters/pulse_instruction.py]: Great!

File: qiskit/qobj/converters/pulse_instruction.py [hunk 2]
```diff
@@ -335,6 +335,8 @@ def convert_frame_change(self, instruction):
             instruction (PulseQobjInstruction): frame change qobj
         Returns:
             Schedule: Converted and scheduled Instruction
+        Raises:
```
Comment (taalexander) [inline qiskit/qobj/converters/pulse_instruction.py]: Can we remove this raises since the check has been removed now?

Comment (nkanazawa1989) [reply qiskit/qobj/converters/pulse_instruction.py]: Sorry! I overlooked. thanks.

**Pull request merged**